### ISSUE TITLE
[no-unused-vars] allow ignore patterns for rest siblings

### ISSUE THREAD

**Issue by jin60641** [typescript-eslint/typescript-eslint#1657]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-unused-vars': ['error', {
      'varsIgnorePattern': '^_',
      'ignoreRestSiblings': false,
    }],
  }
}
```

```TS
// your repro code case
const arr: number[] = [1, 2, 3, 4];
const [_WILL_NOT_USE, willUse1, ...willUse2] = arr;
console.log(willUse2);
```

**Expected Result**
> 'willUse1' is assigned a value but never used  @typescript-eslint/no-unused-vars

**Actual Result**
> '_WILL_NOT_USE' is assigned a value but never used  @typescript-eslint/no-unused-vars
> 'willUse1' is assigned a value but never used  @typescript-eslint/no-unused-vars

**Additional Info**
Change `varsIgnorePattern` option to support rest siblings.
_OR_
Add `restSiblingsIgnorePattern` option.

Thanks

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.11.0` |
| `@typescript-eslint/parser`        | `2.9.0` |
| `TypeScript`                       | `3.7.2` |
| `ESLint`                           | `6.7.1` |
| `node`                             | `12.13.1` |
| `npm`                              | `6.12.1` |

**Comment by bradzacher** [comment]
There's no reason to have this.
Array patterns let you just not specify the element if you don't want to use it.

I.e. this is valid code.
```ts
const [_WILL_NOT_USE, /*unused*/, ...willUse2] = arr;
```

---

Regardless, this is an extension rule, built on top of a base eslint rule. We just add support for TS syntax, and try not to deviate too much from the rule.
Please file a task with them if you'd like to see this added.
